lib/doctor/DiagnosticInfoBuilder.ts

```diff
@@ -0,0 +1,206 @@
+import { Environment } from '@hubspot/local-dev-lib/types/Config';
+import {
+  AccessToken,
+  AccountType,
+  AuthType,
+} from '@hubspot/local-dev-lib/types/Accounts';
+import { Project } from '@hubspot/local-dev-lib/types/Project';
+import { getAccountId } from '../commonOpts';
+import { getAccountConfig, getConfigPath } from '@hubspot/local-dev-lib/config';
+import { getAccessToken } from '@hubspot/local-dev-lib/personalAccessKey';
+import { walk } from '@hubspot/local-dev-lib/fs';
+import util from 'util';
+import { exec as execAsync } from 'child_process';
+import process from 'process';
+
+export type ProjectConfig = Awaited<ReturnType<typeof getProjectConfig>>;
+
+interface FilesInfo {
+  files: string[];
+  configFiles: string[];
+  packageFiles: string[];
+  packageLockFiles: string[];
+  envFiles: string[];
+  jsonFiles: string[];
+}
+
+export interface DiagnosticInfo extends FilesInfo {
+  path?: string;
+  versions: { '@hubspot/cli': string; node: string; npm: string | null };
+  config: string | null;
+  project: {
+    details?: Project;
+    config?: ProjectConfig;
+  };
+  arch: typeof process.arch;
+  platform: typeof process.platform;
+  account: {
+    name?: string;
+    accountId?: number | null;
+    scopeGroups?: AccessToken['scopeGroups'];
+    enabledFeatures?: AccessToken['enabledFeatures'];
+    accountType?: AccessToken['accountType'];
+    authType?: AuthType;
+  };
+  diagnosis?: string;
+}
+
+const configFiles = [
+  'serverless.json',
+  'hsproject.json',
+  'app.json',
+  'public-app.json',
+];
+
+export class DiagnosticInfoBuilder {
+  accountId: number | null;
+  readonly env: Environment | undefined;
+  readonly authType: AuthType | undefined;
+  readonly accountType: AccountType | undefined;
+  readonly personalAccessKey: string | undefined;
+  private _projectConfig?: ProjectConfig;
+  private accessToken?: AccessToken;
+  private projectDetails?: Project;
+  private files?: string[];
+  readonly processInfo: NodeJS.Process;
+
+  constructor(processInfo: NodeJS.Process) {
+    this.accountId = getAccountId();
+    const accountConfig = getAccountConfig(this.accountId!);
+    this.env = accountConfig?.env;
+    this.authType = accountConfig?.authType;
+    this.accountType = accountConfig?.accountType;
+    this.personalAccessKey = accountConfig?.personalAccessKey;
+    this.processInfo = processInfo;
+  }
+
+  async generateDiagnosticInfo(): Promise<DiagnosticInfo> {
+    this._projectConfig = await getProjectConfig(null);
+
+    if (this._projectConfig?.projectConfig) {
+      await this.fetchProjectDetails();
+      await this.fetchAccessToken();
+      await this.fetchProjectFilenames();
+    }
+
+    const {
+      platform,
+      arch,
+      versions: { node },
+      mainModule,
+    } = this.processInfo;
+
+    return {
+      platform,
+      arch,
+      path: mainModule?.path,
+      config: getConfigPath(),
+      versions: {
+        '@hubspot/cli': pkg.version,
```

Comment:
Not a blocker, but we reference `@hubspot/cli` in enough places that it might be worthwhile to add it as a constant somewhere